src/sparse.h

```diff
@@ -146,8 +146,8 @@ class SparseIndices {
         data_.size() / sizeof(int32_t));
   }
 
-  inline void Add(int p, int q, bool seq = false) {
-    data_.extend(sizeof(int64_t), seq);
+  inline void Add(int p, int q) {
```

Comment:
Removing `SparseIndices` entirely sounds great - anything that results in fewer lines of code overall is likely worthwhile.